### ISSUE TITLE
optimizer params restore does not work

### ISSUE THREAD

**Issue by zhengzx-nlp** [whr94621/Transformer-pytorch#8]
learning rate is not yet restorable.

**Comment by whr94621** [comment]
```
optim.optim.load_state_dict(defaultdict(dict, **optimizer_params))
```
I think [this line](https://github.com/whr94621/Transformer-pytorch/blob/d39f39a3996d89f1468d83a6203f0791dea4fbf2/src/main.py#L366) could retore the learning rate ?

**Comment by zhengzx-nlp** [comment]
emmm. So bad things must happen to optim params restore itself
